tensorflow_probability/python/distributions/pert_test.py

```diff
@@ -0,0 +1,205 @@
+"""Tests for PERT distribution."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+
+# Dependency imports
+import numpy as np
+from scipy import stats as sp_stats
+
+import tensorflow.compat.v2 as tf
+import tensorflow_probability as tfp
+
+from tensorflow_probability.python.internal import test_case
+from tensorflow_probability.python.internal import test_util as tfp_test_util
+from tensorflow.python.framework import test_util  # pylint: disable=g-direct-tensorflow-import
+
+tfd = tfp.distributions
+
+@test_util.run_all_in_graph_and_eager_modes
+class PERTTest(test_case.TestCase):
+  def _generate_boilerplate_param(self):
+    smoothness = np.array([1., 2., 4., 10.])
+    mini = np.array(len(smoothness)*[1.])
+    mode = np.array(len(smoothness)*[7.])
+    maxi = np.array(len(smoothness)*[10.])
+    a = 1. + smoothness * (mode - mini) / (maxi - mini)
+    b = 1. + smoothness * (maxi - mode) / (maxi - mini)
+    return smoothness, mini, mode, maxi, a, b
+
+  # Shape and broadcast testing
+  def testPertShape(self):
+    dist = tfd.PERT(mini=[3.0], mode=[10.0], maxi=[11.0], smoothness=[4.0])
+    self.assertEqual(([1]), self.evaluate(dist.batch_shape_tensor()))
+    self.assertEqual(tf.TensorShape([1]), dist.batch_shape)
+    self.assertAllEqual([], self.evaluate(dist.event_shape_tensor()))
+    self.assertEqual(tf.TensorShape([]), dist.event_shape)
+
+  def testBroadcastingSmoothness(self):
+    dist = tfd.PERT(
+        mini=1.,
+        mode=2.,
+        maxi=3.,
+        smoothness=[1., 4., 10.])
+    self.assertEqual([3], self.evaluate(dist.batch_shape_tensor()))
+    self.assertEqual(tf.TensorShape([3]), dist.batch_shape)
+    self.assertAllEqual([], self.evaluate(dist.event_shape_tensor()))
+    self.assertEqual(tf.TensorShape([]), dist.event_shape)
+
+  def testBroadcastingParam(self):
+    dist = tfd.PERT(
+        mini=1.,
+        mode=[2., 3., 4., 5., 6., 7., 8., 9.],
+        maxi=10.,
+        smoothness=4.)
+    self.assertEqual([8], self.evaluate(dist.batch_shape_tensor()))
+    self.assertEqual(tf.TensorShape([8]), dist.batch_shape)
+    self.assertAllEqual([], self.evaluate(dist.event_shape_tensor()))
+    self.assertEqual(tf.TensorShape([]), dist.event_shape)
+
+  def testBroadcastingHigherDimParam(self):
+    dist = tfd.PERT(
+        mini=[1., 2.],
+        mode=[[2., 3.], [4., 5.]],
+        maxi=10.,
+        smoothness=4.,
```

Comment:
I'm guessing we might want to rename this to `temperature`, but if you want you can leave it as is for now, I'll discuss w/ the team when we get ready to merge it.

Reply:
To be honest either works for me intuitively, but I've seen the term `temperature` several times in context of statistics so I changed it.